### PR TITLE
[MPP-1708] Popup Refactor - News Panel

### PR DESCRIPTION
**Pull request by maxxcrawford**
## Summary

Links:
- Jira Ticket: https://mozilla-hub.atlassian.net/browse/MPP-1769
- Epic PR #456 

TODO:
- [x] Dynamical build news list (based on list, audience criteria)
- [x] Dynamical build news detail (story) view
- [x] ~Add "unread' number to icon/logic to control it~
- [x] Add previous news items

Caveats:
- There is a bug where the order of the news items is not set but the content. I'll need to debug. 
- There's currently no logic set to only show to specific audiences or locales. (Everyone gets the same prompts)  

Acceptance Criteria: 
- [x] When there is a new feature show a yellow indicator with a count of the new features.
- [x] Once user has seen them all just show place holder screen
- [x] News and history area with toggle 
- [x] If a user click on a new item then we show a larger panel

## Testing

1. Open news item 
- Run add-on
- Log in
- Open panel > Click news
- **Expected:**  You should have two news items (Bundle, Phones) 
- Click on news item
- **Expected:** New item detail should open up
- Click back button 
- **Expected:** You should now be back at all news panel 

2. Add waffle flag, confirm news item exists 
- Add waffle flag to local Relay instance: `firefox_integration` _(Do this before running the add-on)_
- Run add-on
- Log in
- Open panel > Click news
- **Expected:**  You should have three news items (Bundle, Phones and Firefox Integration) 

## Screenshots

image

image

### CODE REVIEW

File: src/js/popup/popup-new.js [hunk 1]
```diff
@@ -7,30 +7,128 @@
...
+
+  const hasPhone = (await browser.storage.local.get("has_phone")).has_phone;
+  const hasVpn = (await browser.storage.local.get("has_vpn")).has_vpn;
+
+  // Conditions for phone masking announcement to be shown: if the user is in US/CAN, phone flag is on, and user has not purchased phone plan yet
+  const isPhoneMaskingAvailable = isPhoneAvailableInCountry && !hasPhone;
+  // Conditions for bundle announcement to be shown: if the user is in US/CAN, bundle flag is on, and user has not purchased bundle plan yet
+  const isBundleAvailable = isBundleAvailableInCountry && !hasVpn;
+ 
+  // FIXME: The order is not being set correctly
```
Comment (maxxcrawford) [inline src/js/popup/popup-new.js:41]: Issue/Todo: Add date attribute to news items for sorting

File: src/js/popup/popup-new.js [hunk 2]
```diff
@@ -104,79 +221,326 @@
...
+              await browser.storage.local.set({
+                unreadNewsItemsCount: state.newsItemsCount,
+                readNewsItemCount: 0,
+              });
+            }
 
-            // Check if any data applies to the current site
-            const currentPageHostName = await browser.runtime.sendMessage({
-              method: "getCurrentPageHostname",
+            // FIXME: The total news item count may differ than what is displayed to the user
```
Comment (maxxcrawford) [inline src/js/popup/popup-new.js:394]: Known issue/bug.

**Review by maxxcrawford — COMMENTED** [review]
Flagging the FIXMEs

**Review by Vinnl — APPROVED** [review]
Nothing jumped out in a cursory review except that "What’s new in Relay" is unlocalised.

**Pull request merged**